scripts/west/zap_common.py

```diff
@@ -77,13 +81,43 @@ def existing_dir_path(arg: str) -> Path:
     raise argparse.ArgumentTypeError(f'invalid directory path: \'{arg}\'')
 
 
+def update_zcl_in_zap(zap_file: Path, zcl_json: Path):
+    """
+    In the .zap file, there is a relative path to the zcl.json file.
+    Use this function to update zcl.json path if needed.
+    """
+    with open(zap_file, 'r+') as file:
+        data = json.load(file)
+        packages = data.get("package")
+        for package in packages:
+            if package.get("type") == "zcl-properties":
+                package.update({"path": str(zcl_json.absolute().relative_to(zap_file.parent.absolute(), walk_up=True))})
+        file.seek(0)
+        json.dump(data, file, indent=2)
+        file.truncate()
+
+
+def check_zcl_in_zap(zap_file: Path, zcl_json: Path):
+    """
+    Check if the zcl.json path in the .zap file is correct.
+    """
+    with open(zap_file, 'r') as file:
+        data = json.load(file)
+        packages = data.get("package")
+        for package in packages:
+            if package.get("type") == "zcl-properties":
+                return zcl_json.parent.absolute().is_relative_to(zap_file.parent.absolute())
+    return False
+
+
 class ZapInstaller:
     INSTALL_DIR = Path('.zap-install')
     ZAP_URL_PATTERN = 'https://github.com/project-chip/zap/releases/download/v%04d.%02d.%02d-nightly/%s.zip'
 
     def __init__(self, matter_path: Path):
         self.matter_path = matter_path
         self.install_path = matter_path / ZapInstaller.INSTALL_DIR
+        self.cache_path = tempfile.mkdtemp()
```

Comment:
You can use ZAP's `--tempState` instead of `--stateDirectory` with a manually created temp dir.

Reply:
Hmm, oh nice.

Reply:
Done.